src/components/Dialogs/Dialog/Dialog.tsx

```diff
@@ -48,36 +54,39 @@ export const Dialog: React.FC<DialogProps> = (props) => {
 
     return (
         <Modal {...rest} {...block(props)} isOpen={isOpen}>
-            <div {...elem('container')}>
-                {!!title && (
-                    <div {...elem('header')} role="alert">
-                        <Heading level="h3" {...elem('title', props)}>
-                            {title}
-                        </Heading>
-                        <button {...elem('closeButton')} type="button" onClick={onClose}>
-                            <MdClose />
-                        </button>
-                    </div>
-                )}
-                <div ref={ref} {...elem('content')}>
-                    {children}
+            {!!title && closeButton && (
+                <div {...elem('header')} role="alert">
+                    <Heading level="h3" {...elem('title', props)}>
+                        {title}
+                    </Heading>
+                    <Button
+                        {...elem('closeButton')}
+                        variant="ghost"
+                        onClick={closeButton.onClick}
+                        aria-label={closeButton.label}
+                    >
+                        <MdClose />
+                    </Button>
                 </div>
-                {showButtons && (
-                    <div {...elem('actions', { ...props, borderTop: height === 412 })}>
-                        <Button
-                            onClick={acceptButton.onClick}
-                            context={variant === 'primary' ? 'primary' : 'critical'}
-                        >
-                            {acceptButton.label}
-                        </Button>
-                        {!!cancelButton && (
-                            <Button onClick={cancelButton.onClick} variant="ghost">
-                                {cancelButton.label}
-                            </Button>
-                        )}
-                    </div>
-                )}
+            )}
+            <div ref={ref} {...elem('content')}>
+                {children}
             </div>
+            {showButtons && (
+                <div {...elem('actions', { ...props, borderTop: height === 412 })}>
+                    <Button
+                        onClick={acceptButton.onClick}
+                        context={variant === 'primary' ? 'primary' : 'critical'}
+                    >
+                        {acceptButton.label}
+                    </Button>
+                    {!!cancelButton && (
```

Comment:
```suggestion
                    {!!cancelButton.onClick && (
```
I think it is better to check if the function is there, otherwise an empty object is also "true" in JS